cashu/wallet/wallet.py

```diff
@@ -141,22 +178,76 @@ def _construct_proofs(
                 amount=promise.amount,
                 C=C.serialize().hex(),
                 secret=secret,
+                derivation_path=path,
             )
             proofs.append(proof)
+            logger.trace(
+                f"Created proof: {proof}, r: {r.serialize()} out of promise {promise}"
+            )
+
         logger.trace(f"Constructed {len(proofs)} proofs.")
         return proofs
 
+    @staticmethod
+    def _construct_outputs(
+        amounts: List[int], secrets: List[str], rs: List[PrivateKey] = []
+    ) -> Tuple[List[BlindedMessage], List[PrivateKey]]:
+        """Takes a list of amounts and secrets and returns outputs.
+        Outputs are blinded messages `outputs` and blinding factors `rs`
+
+        Args:
+            amounts (List[int]): list of amounts
+            secrets (List[str]): list of secrets
+            rs (List[PrivateKey], optional): list of blinding factors. If not given, `rs` are generated in step1_alice. Defaults to [].
+
+        Returns:
+            List[BlindedMessage]: list of blinded messages that can be sent to the mint
+            List[PrivateKey]: list of blinding factors that can be used to construct proofs after receiving blind signatures from the mint
+
+        Raises:
+            AssertionError: if len(amounts) != len(secrets)
+        """
+        assert len(amounts) == len(
+            secrets
+        ), f"len(amounts)={len(amounts)} not equal to len(secrets)={len(secrets)}"
+        outputs: List[BlindedMessage] = []
+
+        rs_ = [None] * len(amounts) if not rs else rs
+        rs_return: List[PrivateKey] = []
+        for secret, amount, r in zip(secrets, amounts, rs_):
+            B_, r = b_dhke.step1_alice(secret, r or None)
+            rs_return.append(r)
+            output: BlindedMessage = BlindedMessage(
+                amount=amount, B_=B_.serialize().hex()
+            )
+            outputs.append(output)
+        for o, r in zip(outputs, rs_return):
+            logger.trace(f"Constructing output: {o}, r: {r.serialize()}")

```

Comment:
[nitpick] Couldn't you just remove this additional loop and move it to the one above?